test/cpp/qps/perf_db_client.cc

```diff
@@ -0,0 +1,115 @@
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#include "test/cpp/qps/perf_db_client.h"
+
+namespace grpc {
+namespace testing {
+
+//sets the client and server config information
+void PerfDbClient::setConfigs(const ClientConfig& clientConfig, const ServerConfig& serverConfig) {
+  this->clientConfig_ = clientConfig;
+  this->serverConfig_ = serverConfig;
+}
+
+//sets the QPS
+void PerfDbClient::setQPS(double QPS) {
+  this->QPS_ = QPS;
+}
+
+//sets the QPS per core
+void PerfDbClient::setQPSPerCore(double QPSPerCore) {
+  this->QPSPerCore_ = QPSPerCore;
+}
+
+//sets the 50th, 90th, 95th, 99th and 99.9th percentile latency
+void PerfDbClient::setLatencies(double percentileLatency50, double percentileLatency90,
+    double percentileLatency95, double percentileLatency99, double percentileLatency99Point9) {
+  this->percentileLatency50_ = percentileLatency50;
+  this->percentileLatency90_ = percentileLatency90;
+  this->percentileLatency95_ = percentileLatency95;
+  this->percentileLatency99_ = percentileLatency99;
+  this->percentileLatency99Point9_ = percentileLatency99Point9;
+}
+
+//sets the server and client, user and system times
+void PerfDbClient::setTimes(double serverSystemTime, double serverUserTime, 
+    double clientSystemTime, double clientUserTime) {
+  this->serverSystemTime_ = serverSystemTime;
+  this->serverUserTime_ = serverUserTime;
+  this->clientSystemTime_ = clientSystemTime;
+  this->clientUserTime_ = clientUserTime;
+}
+
+//sends the data to the performancew database server
+int PerfDbClient::sendData(std::string access_token, std::string test_name, std::string sys_info, std::string tag) {
+  //Data record request object
+  SingleUserRecordRequest singleUserRecordRequest;
+
+  //setting access token, name of the test and the system information
+  singleUserRecordRequest.set_access_token(access_token);
+  singleUserRecordRequest.set_test_name(test_name);
+  singleUserRecordRequest.set_sys_info(sys_info);
+  singleUserRecordRequest.set_tag(tag);
+
+  //setting configs
+  *(singleUserRecordRequest.mutable_client_config()) = this->clientConfig_;
+  *(singleUserRecordRequest.mutable_server_config()) = this->serverConfig_;
+  
+  Metrics* metrics = singleUserRecordRequest.mutable_metrics();
+
+  //setting metrcs in data record request
```

Comment:
Please  run clang-format, I don't think this looks right.... Also, I would suggest consistently using
if (x) {
 y;
}

Even if y is just a single statement....